### PR TITLE
fixed 1 small bugs in RTM

### PR DESCRIPTION
**Pull request by maaktweluit**
- subtask_id was not used in the loop of discard
- ~subtasks status was not updated on timeout~ ( bigger then expected, made an issue on CLAY )

### CODE REVIEW

File: golem/task/requestedtaskmanager.py [hunk 1]
```diff
@@ -483,13 +483,16 @@ def _get_task_api_service(
...
-    def _check_task_timeout(task_id: TaskId) -> None:
+    def _check_task_timeout(self, task_id: TaskId) -> None:
         task = RequestedTask.get(RequestedTask.task_id == task_id)
         if task.status.is_active():
             logger.info("Task timed out. task_id=%r", task_id)
             task.status = TaskStatus.timeout
             task.save()
+            pending_subtasks = self._get_pending_subtasks(task_id)
+            for subtask in pending_subtasks:
+                subtask.status = SubtaskStatus.failed
```
Comment (Krigpl) [inline golem/task/requestedtaskmanager.py]: Please note that the we may receive the results from the provider anyway and they will handled thus overriding the status to success (I think).

**Review by Krigpl — APPROVED** [review]
A good practice when fixing bugs is to write a unit test to prevent them from happening ever again :)

File: golem/task/requestedtaskmanager.py (hunk 1 above)
Comment (maaktweluit) [reply golem/task/requestedtaskmanager.py]: meh, lets not open this can of worms now ;)

Made an issue to look into this properly

**Comment by maaktweluit** [comment]
@Krigpl good one, extended the unit test to use multiple subtasks to catch this

**Review by Krigpl — APPROVED** [review]
Awesome, thanks for fixing this.

**Pull request merged**